tests/pfc/test_pfc_unpause.py

```diff
@@ -0,0 +1,196 @@
+    )
+
+    flows.append(
+        flow.Flow(
+            name='rx_pfc_unpause',
+            tx_rx=flow.TxRx(
+                flow.PortTxRx(tx_port_name=rx.name, rx_port_name=tx.name)
+            ),
+            packet=[
+                flow.Header(
+                    flow.PfcPause(
+                        src=flow.Pattern('00:AB:BC:AB:BC:AB'),
+                        class_enable_vector=flow.Pattern('FF'),
+                        control_op_code=flow.Pattern('0101'),
+                        pause_class_3=flow.Pattern('0000'),
+                        pause_class_4=flow.Pattern('0000')
+                    )
+                )
+            ],
+            duration=flow.Duration(flow.FixedSeconds(
+                seconds=10, delay=(10**9) * 10, delay_unit='nanoseconds'
+            )),
+            size=flow.Size(size),
+            rate=flow.Rate(value=packets / 10, unit='pps')
+        )
+    )
+
+    cfg = config.Config(
+        ports=[tx, rx], layer1=[tx_l1, rx_l1], flows=flows,
+        options=config.Options(port.Options(location_preemption=True))
+    )
+    utils.start_traffic(api, cfg)
+
+    utils.wait_for(
+        lambda: results_ok(
+            api, utils, packets, lossless_priorities
+        ),
+        'stats to be as expected', timeout_seconds=10
+    )
+
+    utils.wait_for(
+        lambda: results_ok(api, utils, packets, lossless_priorities, False),
+        'stats to be as expected', timeout_seconds=20
+    )
+
+
+def results_ok(api, utils, packets, lossless_priorities, check_for_pause=True):
+    """
+    Returns true if stats are as expected, false otherwise.
+    """
+    port_results = api.get_port_results(utils.result.PortRequest())
+    flow_results = api.get_flow_results(utils.result.FlowRequest())
+    pause = [
+        f['frames_rx'] for f in flow_results if f['name'] == 'rx_pfc_pause'
+    ][0]
+    un_pause = [
+        f['frames_rx'] for f in flow_results if f['name'] == 'rx_pfc_unpause'
+    ][0]
+    ok = []
+
+    for fl in flow_results:
+        if fl['name'] == 'rx_unpause' or fl['name'] == 'rx_pause':
+            continue
+        if pause < packets and pause > 0 and check_for_pause and \
+           "tx_prio_" in fl['name']:
+            ok.append(fl['frames_tx'] == fl['frames_rx'] == 0)
+            continue
+        if un_pause <= packets and un_pause > 0 and \
+           (check_for_pause is False) and "tx_prio_" in fl['name']:
+            ok.append(
+                fl['frames_tx'] == fl['frames_rx'] == packets
+            )
+            continue
+        if "tx_prio_" in fl['name']:
+            ok.append(False)
+    p_stats = {p['name']: p for p in port_results}
+    priorities = [
+        'pfc_class_{}_frames_rx'.format(p) for p in lossless_priorities
+    ]
+    for stat in p_stats['raw_tx']:
+        if (check_for_pause is False) and (stat in priorities):
+            ok.append(
+                p_stats['raw_tx'][stat] == packets * 2
+            )
+
+    return all(ok)
```

Comment:
Need to fix logical issue with `ok`. Need to ensure `ok` contains only given number of flags. i.e. I don't want `all(ok)` to return True when it's equal to `[True]`.

Reply:
made to fixed number of flags